### PR TITLE
testing/ceph: new aport

### PR DESCRIPTION
**Pull request by dx9**
Sorry please ignore the mailing list request. I didn't realize GitHub was an option.

### CODE REVIEW

File: testing/ceph/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+
+package() {
+	cd "$_builddir"
+	cd build
+	make DESTDIR="$pkgdir" install
+	cd ..
+
+	rm -f "$pkgdir"$_sysconfdir/init.d/ceph
+
+	install -D src/etc-rbdmap "$pkgdir"$_sysconfdir/ceph/rbdmap
```
Comment (ktdreyer) [inline testing/ceph/APKBUILD]: "install" installs as executable by default (ie 0755).
Please update this similar to the change in https://github.com/ceph/ceph/pull/11217

Comment (dx9) [inline testing/ceph/APKBUILD]: will do @ktdreyer tks for the heads up.

**Comment by dx9** [comment]
Updated to the 11.0.2 release. Please let me know if there are changes required. Thanks.

File: testing/ceph/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+	mkdir -p "$subpkgdir"$_bindir
+	mv "$pkgdir"$_bindir/rbd-mirror "$subpkgdir"$_bindir
+}
+
+rbd_nbd() {
+	pkgdesc="NBD based client to map Ceph rbd images to local device."
+	depends="librbd librados"
+	mkdir -p "$subpkgdir"$_bindir
+	mv "$pkgdir"$_bindir/rbd-nbd "$subpkgdir"$_bindir
+}
```
Comment (clandmeter) [inline testing/ceph/APKBUILD]: I didn't look into your apkbuild too deeply, but you could try to limited the amount of repeating code by using a general function that accepts some options to split your packages.

File: testing/ceph/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,554 @@
...
+
+librbd() {
+	pkgdesc="RADOS block device client library"
+	depends="librados"
+	mkdir -p "$subpkgdir"/usr/lib
+	mv "$pkgdir"/usr/lib/librbd.so.* "$subpkgdir"/usr/lib
+
+	# TODO: look into this
+	#mkdir -p /usr/lib64/qemu/
+	#ln -sf %{_libdir}/librbd.so.1 /usr/lib64/qemu/librbd.so.1
```
Comment (clandmeter) [inline testing/ceph/APKBUILD]: this looks like copied from fedora? not sure we need that as we do not Multilib.

File: testing/ceph/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,554 @@
...
+py_cephfs() {
+	pkgdesc="Python libraries for Ceph distributed file system"
+	depends="libcephfs py-rados"
+	mkdir -p "$subpkgdir"$_python_sitelib
+	mv "$pkgdir"$_python_sitelib/cephfs.so "$subpkgdir"$_python_sitelib
+	mv "$pkgdir"$_python_sitelib/cephfs-*.egg-info "$subpkgdir"$_python_sitelib
+	mv "$pkgdir"$_python_sitelib/ceph_volume_client.py* "$subpkgdir"$_python_sitelib
+}
+
+ceph_test() {
```
Comment (clandmeter) [inline testing/ceph/APKBUILD]: no pkgdesc?

File: testing/ceph/ceph-common.post-deinstall [hunk 5]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+rm -rf /var/log/ceph
+rm -rf /etc/ceph
```
Comment (clandmeter) [inline testing/ceph/ceph-common.post-deinstall]: I don't think we want to do this. Let the user cleanup their own non apk files.

File: testing/ceph/APKBUILD [hunk 6]
```diff
@@ -0,0 +1,554 @@
...
+	mv "$pkgdir"$_bindir/ceph_smalliobenchrbd "$subpkgdir"$_bindir
+	mv "$pkgdir"$_bindir/ceph_test_* "$subpkgdir"$_bindir
+	mv "$pkgdir"$_bindir/ceph_tpbench "$subpkgdir"$_bindir
+	mv "$pkgdir"$_bindir/ceph_xattr_bench "$subpkgdir"$_bindir
+	mv "$pkgdir"$_bindir/ceph-coverage "$subpkgdir"$_bindir
+	mv "$pkgdir"$_bindir/ceph-monstore-tool "$subpkgdir"$_bindir
+	mv "$pkgdir"$_bindir/ceph-osdomap-tool "$subpkgdir"$_bindir
+	mv "$pkgdir"$_bindir/ceph-kvstore-tool "$subpkgdir"$_bindir
+	mv "$pkgdir"$_bindir/ceph-debugpack "$subpkgdir"$_bindir
+	mv "$pkgdir"$_bindir/ceph-client-debug "$subpkgdir"$_bindir
```
Comment (clandmeter) [inline testing/ceph/APKBUILD]: a simple loop could reduce this to ~4 lines.

File: testing/ceph/APKBUILD [hunk 7]
```diff
@@ -0,0 +1,554 @@
...
+depends="
+	ceph-osd
+	ceph-mds
+	ceph-mon
+	binutils
+"
+
+makedepends="
+	autoconf
+	automake
```
Comment (dx9) [inline testing/ceph/APKBUILD]: nope. cruft from before the switch to cmake. I'll delete them

File: testing/ceph/APKBUILD [hunk 8]
```diff
@@ -0,0 +1,554 @@
...
+	acl-dev
+	argp-standalone
+	binutils
```
Comment (dx9) [inline testing/ceph/APKBUILD]: yep, I'll delete

File: testing/ceph/APKBUILD (hunk 4 above)
Comment (dx9) [inline testing/ceph/APKBUILD]: will add

File: testing/ceph/ceph-common.post-deinstall (hunk 5 above)
Comment (dx9) [inline testing/ceph/ceph-common.post-deinstall]: ok

File: testing/ceph/APKBUILD (hunk 3 above)
Comment (dx9) [inline testing/ceph/APKBUILD]: yeah this whole scheme is based on https://github.com/dx9/ceph/blob/master/ceph.spec.in. I'll just toss this

File: testing/ceph/APKBUILD [hunk 9]
```diff
@@ -0,0 +1,554 @@
...
+"
+
+case $pkgrel in
+	''|*[!0-9]*)
+		_builddir="$srcdir"/$pkgname-$pkgver-`echo $pkgrel | sed 's/\./-/'`
+		_sourcepkg=ceph-$pkgver-`echo $pkgrel | sed 's/\./-/'`.tar.bz2
+		;;
+	*)
+		_builddir="$srcdir"/$pkgname-$pkgver
+		_sourcepkg="http://github.com/dx9/ceph/releases/download/$pkgver/$pkgname-$pkgver.tar.bz2"
```
Comment (dx9) [inline testing/ceph/APKBUILD]: not yet. was still waiting on a few merges. it's ready to move to official release with upcoming 11.1.0

File: testing/ceph/APKBUILD [hunk 10]
```diff
@@ -0,0 +1,554 @@
...
+	cython-dev
+	diffutils
+	eudev-dev
+	expat-dev
+	fcgi-dev
+	findutils
+	flex
+	fuse-dev
+	git
+	grep
```
Comment (dx9) [inline testing/ceph/APKBUILD]: pretty sure it was an unsupported option, I'll retest

File: testing/ceph/APKBUILD [hunk 11]
```diff
@@ -0,0 +1,554 @@
...
+	parted
+	procps
+	python-dev
+	py-pip
+	py-nose
+	py-sphinx
+	py-virtualenv
+	readline-dev
+	rpcgen
+	sed
```
Comment (dx9) [inline testing/ceph/APKBUILD]: same as sed, I'll retest

File: testing/ceph/APKBUILD [hunk 12]
```diff
@@ -0,0 +1,554 @@
...
+	autoconf
+	automake
+	acl-dev
+	argp-standalone
+	binutils
+	boost-dev
+	btrfs-progs
+	build-base
+	bzip2-dev
+	ccache
```
Comment (dx9) [inline testing/ceph/APKBUILD]: Nope, it's optional. So ya, will remove.

File: testing/ceph/APKBUILD [hunk 13]
```diff
@@ -0,0 +1,554 @@
...
+
+makedepends="
+	autoconf
+	automake
+	acl-dev
+	argp-standalone
+	binutils
+	boost-dev
+	btrfs-progs
+	build-base
```
Comment (dx9) [inline testing/ceph/APKBUILD]: right, I'll remove

File: testing/ceph/APKBUILD [hunk 14]
```diff
@@ -0,0 +1,554 @@
...
+	install -m 0644 -D udev/95-ceph-osd.rules "$pkgdir"$_udevrulesdir/95-ceph-osd.rules
+}
+
+base() {
+	pkgdesc="Base is the package that includes all the files shared amongst ceph servers"
+	depends="
+		ceph-common
+		cryptsetup
+		findutils
+		grep
```
Comment (dx9) [inline testing/ceph/APKBUILD]: ummm not sure... I'll take a closer look.

File: testing/ceph/APKBUILD [hunk 15]
```diff
@@ -0,0 +1,554 @@
...
+		_builddir="$srcdir"/$pkgname-$pkgver-`echo $pkgrel | sed 's/\./-/'`
+		_sourcepkg=ceph-$pkgver-`echo $pkgrel | sed 's/\./-/'`.tar.bz2
+		;;
+	*)
+		_builddir="$srcdir"/$pkgname-$pkgver
+		_sourcepkg="http://github.com/dx9/ceph/releases/download/$pkgver/$pkgname-$pkgver.tar.bz2"
+		;;
+esac
+
+#source="https://github.com/ceph/ceph/archive/v$pkgver.zip"
```
Comment (dx9) [inline testing/ceph/APKBUILD]: 👍

File: testing/ceph/APKBUILD (hunk 10 above)
Comment (dx9) [inline testing/ceph/APKBUILD]: grep --quiet usage will add comment

File: testing/ceph/APKBUILD (hunk 6 above)
Comment (dx9) [inline testing/ceph/APKBUILD]: yeah I'll clean these up

File: testing/ceph/APKBUILD [hunk 16]
```diff
@@ -0,0 +1,554 @@
...
+	sed
+	snappy-dev
+	util-linux
+	userspace-rcu-dev
+	xfsprogs-dev
+	xmlstarlet
+	yasm
+"
+
+case $pkgrel in
```
Comment (dx9) [inline testing/ceph/APKBUILD]: ya, trying to integrate with the way the ceph tree works. I'll get rid of this.

File: testing/ceph/APKBUILD (hunk 2 above)
Comment (dx9) [inline testing/ceph/APKBUILD]: 👍

**Comment by clandmeter** [comment]
Be accident (because this was also posted on the mailing list) this has been committed already to aports.
Some of the requested changes have been made, but I would still like you to submit a new PR and fix a few of the other outstanding issues for instance the upstream source part.

**Comment by dx9** [comment]
@clandmeter Yeah for sure. I've been working on an update to the 11.1.0 release with all the requested changes. Sorry for the confusion and thanks for all your feedback.

**Pull request closed without merging**